Source/Core/DolphinTool/ConvertCommand.cpp

```diff
@@ -0,0 +1,281 @@
+    return 1;
+  }
+  const DiscIO::BlobType format = format_o.value();
+
+  // Open the volume now for inspection
+  std::unique_ptr<DiscIO::Volume> volume = DiscIO::CreateVolume(input_file_path);
+  if (!volume)
+  {
+    std::cerr << "Error: Unable to open disc image" << std::endl;
+    return 1;
+  }
+
+  // --scrub
+  const bool scrub = static_cast<bool>(options.get("scrub"));
+
+  if (scrub && (format == DiscIO::BlobType::RVZ || volume->IsDatelDisc()))
+  {
+    std::cerr << "Error: Scrubbing an RVZ container or Datel disc is not supported";
+    return 1;
+  }
+
+  if (scrub && format == DiscIO::BlobType::PLAIN)
+  {
+    std::cerr << "Warning: Scrubbing does not save space when converting to ISO unless using "
+                 "external compression."
+              << std::endl;
+  }
+
+  if (!scrub && format == DiscIO::BlobType::GCZ &&
+      volume->GetVolumeType() == DiscIO::Platform::WiiDisc && !volume->IsDatelDisc())
+  {
+    std::cerr << "Warning: Converting Wii disc images to GCZ without scrubbing may not offer space "
+                 "advantages over ISO."
+              << std::endl;
+  }
+
+  if (volume->IsNKit())
+    std::cerr << "Warning: Converting an NKit file, output will still be NKit!" << std::endl;
+
+  // --block_size
+  std::optional<int> block_size_o;
+  if (options.is_set("block_size"))
+    block_size_o = static_cast<int>(options.get("block_size"));
+
+  if (format == DiscIO::BlobType::GCZ || format == DiscIO::BlobType::WIA ||
+      format == DiscIO::BlobType::RVZ)
+  {
+    if (!block_size_o.has_value())
+    {
+      std::cerr << "Error: Block size must be set for GCZ/RVZ/WIA" << std::endl;
+      return 1;
+    }
+
+    if (!DiscIO::IsDiscImageBlockSizeCompatible(block_size_o.value(), format))
+    {
+      std::cerr << "Error: Block size is invalid" << std::endl;
+      return 1;
+    }
+  }
+
+  if (DiscIO::IsGCZBlockSizeLegacyCompatible(block_size_o.value(), volume->GetSize()))
+  {
+    std::cerr << "Warning: For GCZs, the file size must be an integer multiple of "
+                 "the block size "
+                 "and must not be an integer multiple of the block size multiplied by 32.";
+  }
+
+  // --compress, --compress_level
+  std::optional<DiscIO::WIARVZCompressionType> compression_o =
+      ParseCompressionTypeString(static_cast<const char*>(options.get("compression")));
+
+  std::optional<int> compression_level_o;
+  if (options.is_set("compression_level"))
+    compression_level_o = static_cast<int>(options.get("compression_level"));
+
+  if (format == DiscIO::BlobType::WIA || format == DiscIO::BlobType::RVZ)
+  {
+    if (!compression_o.has_value())
+    {
+      std::cerr << "Error: Compression format must be set for WIA or RVZ" << std::endl;
+      return 1;
+    }
+
+    if (compression_o.value() == DiscIO::WIARVZCompressionType::Zstd &&
+        format == DiscIO::BlobType::WIA)
+    {
+      std::cerr << "Error: WIA does not support Zstd" << std::endl;
+      return 1;
+    }
```

Comment:
`ParseCompressionTypeString` can return `DiscIO::WIARVZCompressionType::Purge`. You'll need to reject `DiscIO::WIARVZCompressionType::Purge` for RVZ.